### DISCUSSION TITLE
How to plot a line through a 3D object in pyvista in the direction of vectors?

### DISCUSSION THREAD

**Discussion by harshamarupudi56** [pyvista/pyvista#3043] [Q&A] [answered]
I have a numpy array of eigenvectors corresponding to a file that pyvista is reading. I want to plot a line through the 3D object in the direction of those vectors? 

I am attempting to do 
```py
eigenvectors = np.array[1,2,3,4,5,6]
mesh1 = pv.read("file.vtk")
plotter = pv.Plotter

plotter.add_mesh(mesh1)
plotter.add_mesh(eigenvectors, 'r','wireframe')
```
Do I need to convert the numpy array into a vtk file or mesh as well before I plot it? With the way I have it set up now I think I will just end up overlaying the vectors over the mesh1 but not the plot a line through the center to see the direction of the volume.

**Comment by akaszynski** [comment]
> I want to plot a line through the 3D object in the direction of those vectors?

You can add the eigenvectors as scalars that are associated with the mesh. We have some eigenplotting examples that you might want to take a look at here: https://docs.pyvista.org/examples/99-advanced/warp-by-vector-eigenmodes.html

Your code will then look like:

```py
mesh1 = pv.read("file.vtk")
plotter = pv.Plotter()
plotter.add_mesh(mesh1, scalars=eigenvectors)
plotter.show()
```

**Comment by harshamarupudi56** [comment, marked as answer]
Not quite what I was looking for. 

But I managed to do what I wanted by plotting an arrow in the direction of the eigenvectors. 

```py
arrow = pyvista.Arrow(direction = eigenvectors)
plotter.add_mesh(mesh1)
plotter.add_mesh(arrow)
```

Thanks for help!
